web/src/components/DatePicker.tsx

```diff
@@ -132,34 +89,35 @@ const CustomDatePicker = ({
 
   return (
     <DateContainer>
-      <StyledInputWrapper>
-        <StyledInput
-          open={isCalendarOpen}
-          value={value}
-          popperPlacement='bottom'
-          selected={isValidJsDate(date?.toJSDate() ?? null) ? date?.toJSDate() : null}
-          onSelect={e => {
-            setValue(DateTime.fromJSDate(e ?? new Date()).toFormat('dd.MM.yyyy'))
-            setDatePickerError('')
+      <LocalizationProvider dateAdapter={CustomAdapterLuxon} adapterLocale={languageCode}>
+        <DatePicker
+          value={date}
+          onChange={newValue => {
+            if (newValue) {
+              handleDateChange(newValue.toJSDate())
+              setValue(newValue.toFormat('dd.MM.yyyy'))
+              setDatePickerError('')
+            }
+          }}
+          slotProps={{
+            textField: {
+              placeholder: placeholderDate
+                .setLocale(languageCode)
+                .toLocaleString({ month: '2-digit', day: '2-digit', year: 'numeric' }),
+            },
+            openPickerButton: {
+              'aria-label': calendarLabel,
+            },
+          }}
+          onError={reason => {
+            if (reason) {
```

Comment:
🔧 Why is this if necessary? I think `reason` is always truthy.

Reply:
Oh okay, seems like the typing is wrong. I guess it would still be better practice to just save the error/reason in the state and then decide separately what to do with the specific error:
``` tsx
const [validationError, setValidationError] = React.useState<DateValidationError | null>(null)
const errorMessage = error ?? (validationError ? t('invalidDate') : null)

<DatePicker onError={setError} ... />
```